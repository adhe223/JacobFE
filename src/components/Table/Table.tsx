import * as React from 'react';

// TableRow looks like
// {
//   'Name': 'Alex',
//   'id': '1',
//   'data': 'science',
//    ...: ...,
// }
export interface TableRow {
  [key: string]: string;
}

export interface TableProps {
  columnHeadings: string[];
  listItems: TableRow[];
}

const Table = (props: TableProps) => {
  const headings = props.columnHeadings.map((heading, indx) => {
    return <th key={'th-' + indx}>{heading}</th>;
  });

  const rows = props.listItems.map((row, rowIndx) => {
    let cells: JSX.Element[] = [];

    Object.keys(row).forEach((heading, cellIndx) => {
      cells.push(
        <td key={'td-' + cellIndx}>{row[heading]}</td>
      );
    });

    return <tr key={'tr-' + rowIndx}>{cells}</tr>;
  });

  return (
    <table>
      <tr>
        {headings}
      </tr>
      {rows}
    </table>
  );
};

export default Table;