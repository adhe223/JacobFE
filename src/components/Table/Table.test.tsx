import * as React from 'react';
import Table, { TableProps } from './Table';
import * as renderer from 'react-test-renderer';

const fakeTableProps: TableProps = {
  columnHeadings: ['Name', 'id', 'data'],
  listItems: [{
    'Name': 'Alex',
    'id': '0',
    'data': 'stud',
  }, {
    'Name': 'Cindy',
    'id': '1',
    'data': 'studdette',
  }],
};

describe('Table', () => {
  describe('render', () => {
    const component = renderer.create(
      <Table {...fakeTableProps}/>
    );
    let tree = component.toJSON();

    it('should render correctly', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});