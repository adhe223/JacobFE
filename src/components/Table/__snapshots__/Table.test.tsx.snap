// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table render should render correctly 1`] = `
<table>
  <tr>
    <th>
      Name
    </th>
    <th>
      id
    </th>
    <th>
      data
    </th>
  </tr>
  <tr>
    <td>
      Alex
    </td>
    <td>
      0
    </td>
    <td>
      stud
    </td>
  </tr>
  <tr>
    <td>
      Cindy
    </td>
    <td>
      1
    </td>
    <td>
      studdette
    </td>
  </tr>
</table>
`;
